src/compiler/core.ts

```diff
@@ -633,6 +633,10 @@ module ts {
         }
     }
 
+    export function getDefaultLibraryFilename(options: CompilerOptions): string {
+        return options.target === ScriptTarget.ES6 ? "lib.es6.d.ts" : "lib.d.ts";
+    }
+

```

Comment:
i think we should only have one lib file with some details inside on when declarations are relevant, e.g. decorators/attribute annotations that say if this declaration applies to the current compilation or not.
managing multiple files has been a hassle.